File: coachmark/src/main/java/com/rizafu/coachmark/ViewUtils.java
package com.rizafu.coachmark;

import android.content.res.Resources;
import android.view.View;
import android.view.ViewTreeObserver;

/**
 * Created by RizaFu on 11/12/16.
 */

class ViewUtils {

    public static int pxToDp(int px) {
        return (int) (px / Resources.getSystem().getDisplayMetrics().density);
    }

    static int dpToPx(int dp) {
        return (int) (dp * Resources.getSystem().getDisplayMetrics().density);
    }

    public static void doAfterLayout(final View view, final Runnable runnable) {
        ViewTreeObserver.OnGlobalLayoutListener listener = new ViewTreeObserver.OnGlobalLayoutListener() {
            @Override
            public void onGlobalLayout() {
                ViewTreeObserver viewTreeObserver = view.getViewTreeObserver();
                viewTreeObserver.removeOnGlobalLayoutListener(this);
                runnable.run();
            }
        };
        view.getViewTreeObserver().addOnGlobalLayoutListener(listener);
    }
}

File: coachmark/src/main/java/com/rizafu/coachmark/ViewUtils.kt
package com.rizafu.coachmark

import android.content.res.Resources
import android.view.View
import android.view.ViewTreeObserver

/**
 * Created by RizaFu on 11/12/16.
 */

internal object ViewUtils {

    fun pxToDp(px: Int): Int {
        return (px / Resources.getSystem().displayMetrics.density).toInt()
    }

    fun dpToPx(dp: Int): Int {
        return (dp * Resources.getSystem().displayMetrics.density).toInt()
    }

    fun doAfterLayout(view: View, listen: () -> Unit) {
        val listener = object : ViewTreeObserver.OnGlobalLayoutListener {
            override fun onGlobalLayout() {
                val viewTreeObserver = view.viewTreeObserver
                viewTreeObserver.removeOnGlobalLayoutListener(this)
                listen.invoke()
            }
        }
        view.viewTreeObserver.addOnGlobalLayoutListener(listener)
    }
}
